tests/unit/v1/test_async_transaction.py

```diff
@@ -80,15 +82,10 @@ def test__clean_up(self):
 
     @pytest.mark.asyncio
     async def test__begin(self):
-        from google.cloud.firestore_v1.services.firestore import (
-            client as firestore_client,
-        )
         from google.cloud.firestore_v1.types import firestore
 
         # Create a minimal fake GAPIC with a dummy result.
-        firestore_api = mock.create_autospec(
-            firestore_client.FirestoreClient, instance=True
```

Comment:
So my understanding is that using `create_autospec` implies that the mock's methods will exactly match those of the FirestoreClient, whereas using a generic AsyncMock allows you to call any method, including ones that don't exist on the real client. Any thoughts on whether this will be a reasonable choice for this test (and the ones below with similar changes)?

Reply:
That is a valid point - my thought was that as functionality is only defined for the mocked functions we care about, that autospecing is not entirely necessary. I can see what you're coming from though, and am happy to manually add spec. @crwilcox what are your thoughts?

Reply:
@tritone has it right here. Autospecs will fail if you try to call a method that doesn't match the class it is mocking. On the other hand, it often means more setup, when a generic mock would allow it in less.

I think autospecs can be useful. Is there a reason to move to a generic mock that I am not seeing?

Reply:
We're using the modified mock class (`AsyncMock`) that overloads the `__call__` function with a coroutine. This isn't stopping us from manually creating the spec, but I don't believe we can use autospec.

Reply:
We talked about this a bit. We already use a mix so this matches precedent. I do think autospec may have value but also isn't necessary. I would be satisfied to track this as an issue to dig into later.

Reply:
Created issue #133

Reply:
Sounds good to me; thanks for following up.